downstairs/src/region.rs

```diff
@@ -2180,12 +2236,27 @@ impl Region {
             }
             None => self.def.extent_count().try_into().unwrap(),
         };
+
+        // Spawn parallel tasks for the flush
+        let mut join_handles: Vec<JoinHandle<Result<(), CrucibleError>>> =
+            Vec::with_capacity(extent_count);
+
         for eid in 0..extent_count {
-            let extent = &self.extents[eid];
-            extent
-                .flush(flush_number, gen_number, job_id, &self.log)
-                .await?;
+            let extent = self.get_opened_extent(eid).await;
+            let log = self.log.clone();
+            let jh = tokio::spawn(async move {
+                extent.flush(flush_number, gen_number, job_id, &log).await
+            });
+            join_handles.push(jh);
         }
+
+        // Wait for all flushes to finish
+        for join_handle in join_handles {
+            join_handle
+                .await
+                .map_err(|e| CrucibleError::GenericError(e.to_string()))??;
```

Comment:
If we error here, where does this bubble out to?

Reply:
if any of the flushes fails, then it will get returned as a failure here for `region_flush`. Though, see 53c45fbef40f410eabe218d9df2502a840e8e6f2 for the fix for an issue you brought up during a chat: we should wait for all results before returning, instead of (potentially) cancelling these flush tasks in the middle of execution if one of them errors.